SmartCarMQTTApp/app/src/androidTest/java/com/example/smartcarmqttapp/MqttCarUnitTest.java

```diff
@@ -0,0 +1,111 @@
+package com.example.smartcarmqttapp;
+
+import static org.junit.Assert.*;
+
+import android.content.Context;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.Callable;
+
+@RunWith(AndroidJUnit4.class)
+public class MqttCarUnitTest {
+
+    /**
+     * Intervals in ms.
+     */
+    public static final class Intervals {
+        public static final int Short = 300;
+        public static final int Medium = 500;
+        public static final int Long = 1000;
+    }
+
+    /**
+     * Timeout in ms.
+     */
+    public static final class Timeout {
+        public static final int Short = 1000;
+        public static final int Medium = 5000;
+        public static final int Long = 10000;
+    }
+
+    /**
+     * Fault tolerance
+     */
+    public static final class FaultTolerance {
+        public static final double Speed = 0.01;
+    }
+
+    private static Context appContext;
+    private static MqttCar car;
+
+    @BeforeClass
+    public static void setupCar() throws InterruptedException {
+        appContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
```

Comment:
This feels like it's not a unit test but an instrumented test. Does this test even pass? If it does, why don't you run it on CI?

Reply:
Because, it relies on a car running and responding over mqtt, like explained in #20. The only reason this is an instrumented test is that the mqtt library seems to need a context to work

Reply:
OK, so if I understand this correctly, you are planning to run this manually on your machines while having the emulator environment up and running?

Reply:
Exactly, and in sprint 2 when we start working on the quiz logic, we will test the quiz state management automated in a GitHub action for the course requirements for getting a 5.

Reply:
A wiki page regarding the testing is on my todo list for tonight after I finish my work 😅.